airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/implementor/ProcessRecordsTask.kt

```diff
@@ -37,81 +38,84 @@ interface ProcessRecordsTask : StreamLevel, ImplementorScope
+            log.info { "Processing records from $file for stream $streamDescriptor" }
+            val batch =
+                try {
+                    file.localFile.inputStream().use { inputStream ->
+                        val records =
+                            inputStream
+                                .lineSequence()
+                                .map {
+                                    when (val message = deserializer.deserialize(it)) {
+                                        is DestinationStreamAffinedMessage -> message
+                                        else ->
+                                            throw IllegalStateException(
+                                                "Expected record message, got ${message::class}"
+                                            )
+                                    }
                                 }
-                            }
-                            .takeWhile {
-                                it !is DestinationRecordStreamComplete &&
-                                    it !is DestinationRecordStreamIncomplete
-                            }
-                            .map { it as DestinationRecord }
-                            .iterator()
-                    streamLoader.processRecords(records, file.totalSizeBytes)
+                                .takeWhile {
+                                    it !is DestinationRecordStreamComplete &&
+                                        it !is DestinationRecordStreamIncomplete
+                                }
+                                .map { it as DestinationRecord }
+                                .iterator()
+                        val batch = streamLoader.processRecords(records, file.totalSizeBytes)
+                        log.info { "Finished processing $file" }
+                        batch
+                    }
+                } finally {
+                    log.info { "Processing completed, deleting $file" }
+                    file.localFile.toFile().delete()
+                    diskManager.release(file.totalSizeBytes)
                 }
-            } finally {
-                log.info { "Processing completed, deleting $file" }
-                file.localFile.toFile().delete()
-                diskManager.release(file.totalSizeBytes)
-            }
 
-        val wrapped = BatchEnvelope(batch, file.indexRange)
-        taskLauncher.handleNewBatch(streamDescriptor, wrapped)
+            val wrapped = BatchEnvelope(batch, file.indexRange)
+            log.info { "Updating batch $wrapped for $streamDescriptor" }
+            taskLauncher.handleNewBatch(streamDescriptor, wrapped)
+        }
     }
 }
 
 interface ProcessRecordsTaskFactory {
     fun make(
         taskLauncher: DestinationTaskLauncher,
-        stream: DestinationStream.Descriptor,
-        file: SpilledRawMessagesLocalFile,
     ): ProcessRecordsTask
 }
 
+data class FileAggregateMessage(
```

Comment:
Looking ahead to possible interface changes, if `FileAggregateMessage` extends something like

```
interface Aggregate {
  records: Flow<Record>
}
```

Then the `SpillToDiskTask` can become something like `RecordAggregatorTask`, which gets injected with `RecordAggregator`, which could  be `FileAggregator` (the current behavior) or just a Noop. (Promoting an `Iterator` to a `Flow` is trivial.)

The trigger for that change could be "we want increased performance on the tasks that don't need buffering, like object storage uploads with small part sizes."

Then `ProcessRecordsTask` could become something like `AccumulateBatchTask`, which takes a `BatchAccumulator<T, U: Batch>` where we set up `nProcessRecordsTasks` instances of `BatchAccumulator<Aggregate, [client-defined Batch type]>` and `nProcessBatchTasks` of `BatchAccumulator<[client-defined], [client-defined]>`. (All of which just use the client-provided accumulators.)